packages/go/src/go-helpers.ts

```diff
@@ -426,36 +426,53 @@ class GoError extends Error {
  * @param modulePath The directory containing the `go.mod` file
  * @returns
  */
-async function parseGoModVersion(
-  modulePath: string
-): Promise<string | undefined> {
-  let version;
+async function getFullVersionFromGoMod(modulePath: string) {
   const file = join(modulePath, 'go.mod');
 
+  let content;
   try {
-    const content = await readFile(file, 'utf8');
-    const matches = /^go (\d+)\.(\d+)\.?$/gm.exec(content) || [];
-    const major = parseInt(matches[1], 10);
-    const minor = parseInt(matches[2], 10);
-    const full = versionMap.get(`${major}.${minor}`);
-    if (major === 1 && minor >= GO_MIN_VERSION && full) {
-      version = full;
-    } else if (!isNaN(minor)) {
-      const err = new GoError(
-        `Unsupported Go version ${major}.${minor} in ${file}`
-      );
-      err.code = 'ERR_UNSUPPORTED_GO_VERSION';
-      throw err;
-    } else {
-      console.log(`Warning: Unknown Go version in ${file}`);
-    }
+    content = await readFile(file, 'utf8');
   } catch (err: any) {
     if (err.code === 'ENOENT') {
       debug(`File not found: ${file}`);
+      return undefined;
     } else {
       throw err;
     }
   }
 
-  return version;
+  const version = parseGoVersion(content);
+  if (!version) {
+    console.log(`Warning: Unknown Go version in ${file}`);
+    return undefined;
+  }
+  if (version.major === 1 && version.minor >= GO_MIN_VERSION) {
+    return version.patch === null
+      ? versionMap.get(`${version.major}.${version.minor}`)
+      : `${version.major}.${version.minor}.${version.patch}`;
+  } else {
+    const versionString = version.patch
+      ? `${version.major}.${version.minor}.${version.patch}`
+      : `${version.major}.${version.minor}`;
+    const err = new GoError(
+      `Unsupported Go version ${versionString} in ${file}`
+    );
+    err.code = 'ERR_UNSUPPORTED_GO_VERSION';
+    throw err;
+  }
+}
+
+function parseGoVersion(goModContent: string) {
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** Can we add tests around this?